### PR TITLE
fix(axios-extension): parsing of service gen response

### PR DESCRIPTION
**Pull request by docirl**
https://github.com/SAP/open-ux-tools/issues/2087

Fix parsing of ui service generation response, use `parseResponse` instead of json.parse

### CODE REVIEW

File: packages/axios-extension/src/abap/adt-catalog/generators/ui-service-generator.ts [hunk 1]
```diff
@@ -124,6 +124,6 @@ export class UiServiceGenerator extends AdtService {
                 corrNr: transport
             }
         });
-        return JSON.parse(response.data);
+        return this.parseResponse(response.data);
```
Comment (Klaus-Keller) [inline packages/axios-extension/src/abap/adt-catalog/generators/ui-service-generator.ts:127]: @docirl, in line 120 above we set `Accept` to `+json` and `+xml`. With this change we handle only xml, no json anymore. Are we sure the response is always xml?

Comment (docirl) [reply packages/axios-extension/src/abap/adt-catalog/generators/ui-service-generator.ts:127]: Yes the response is json, and response.data is an xml string.

**Review by devinea — APPROVED** [review]
Code changes and tests ✅ 
changeset ✅

**Review by Klaus-Keller — APPROVED** [review]
Thanks @docirl,
- fix is easy to understand 
- test coverage is great
- did no local test, just visual review

Approved from my side.

**Comment by docirl** [comment]
thanks @devinea @Klaus-Keller

**Pull request merged**